include/range/v3/view/iota.hpp

```diff
@@ -310,10 +328,45 @@ namespace ranges
                               ? finite
                               : unknown>
     {
+
     private:
+        struct step
+        {
+            detail::iota_difference_t<From> value() const
+            {
+                return value_;
+            }
+
+            constexpr operator detail::iota_difference_t<From>() const
+            {
+                return value();
+            }
+
+            constexpr step() = default;
+            constexpr step(detail::iota_difference_t<From> i)
+              : value_(i)
+            {}
+
+            struct empty
+            {
+                constexpr empty() = default;
+                constexpr empty(detail::iota_difference_t<From>)
+                {}
+                constexpr operator detail::iota_difference_t<From>() const
+                {
+                    return 1;
+                }
+            };
+            RANGES_NO_UNIQUE_ADDRESS
+            std::conditional_t<detail::iota_stridable_<From, To>,
+                               detail::iota_difference_t<From>, empty>
+                value_{1};
+        };
+
```

Comment:
This is what `constant` and `mutable_` were for in `<range/v3/utility/box.hpp>`, but they are currently deprecated. Here's what I would suggest: Copy `mutable_` into a new (undeprecated) class `variable` that nukes the `mutable` on the data member and is `const`-correct. Then undeprecate `constant` and use those instead of inventing this here. (Please leave `mutable_` deprecated.)